src/calc-length.js

```diff
@@ -93,6 +93,70 @@
     return new CalcLength(calcDictionary);
   };
 
+  CalcLength.prototype._addCalcLengths = function(addedLength) {
+    if (!(addedLength instanceof CalcLength)) {
+      throw new TypeError('Objects not of type CalcLength');
+    } 
+    
+    var calcDictionary = {};
+
+    // Iterate through all possible length types and add their values
+    for (var i = 0; i < shared.LengthValue.LengthType.length; i++){
+      var type = shared.LengthValue.LengthType[i];
+      if (this[type] == null){
+        calcDictionary[type] = addedLength[type];
+      } else if (addedLength[type] == null) {
+        calcDictionary[type] = this[type];
+      } else {
+        calcDictionary[type] = this[type] + addedLength[type];
+      }
+    }
+
+    return new CalcLength(calcDictionary);
+  };
+
+  CalcLength.prototype._subtractCalcLengths = function(subtractedLength) {
+    if (!(subtractedLength instanceof CalcLength)) {
+      throw new TypeError('Objects not of type CalcLength');
+    } 
+    
+    var calcDictionary = {};
+
+    // Iterate through all possible length types and add their values
+    for (var i = 0; i < shared.LengthValue.LengthType.length; i++){
+      var type = shared.LengthValue.LengthType[i];
+      if (subtractedLength[type] == null) {
+        calcDictionary[type] = this[type];
+      } else if (this[type] == null){
+        calcDictionary[type] = -subtractedLength[type];
+      } else {
+        calcDictionary[type] = this[type] - subtractedLength[type];
+      }
+    }
+
+    return new CalcLength(calcDictionary);
+  };
+
+  LengthValue.prototype._asCalcLength = function() {
+    return this;
+  };
+
+  CalcLength.prototype.equals = function(other) {
+    if (!(other instanceof CalcLength)){
+      return false;
+    }
+
+    // Iterate through all length types and check that both objects contain the same values
+    for (var index in shared.LengthValue.LengthType){
```

Comment:
'index' should just be 'i'. Same elsewhere in this file. Loop counters can be given special names when there's a good reason to do so; otherwise just use i, j, k.